src/basic/fd-util.h

```diff
@@ -52,6 +52,11 @@ static inline void fclosep(FILE **f) {
         safe_fclose(*f);
 }
 
+static inline void *close_fd_ptr(void *p) {
+        safe_close(PTR_TO_FD(p));
+        return NULL;
+}
+
```

Comment:
split this one out too, even if there's only a single user it sounds good enough to make generic.